perfdash/google-gcs-downloader.go

```diff
@@ -182,13 +182,21 @@ func getBuildData(result JobToCategoryData, prefix string, category string, labe
 	return result[prefix][category][label]
 }
 
-type bucketUtil struct {
+// BucketUtil is the interface that fetches data from a storage service
+type BucketUtil interface {
+	getBuildNumbersFromBucket(job string) ([]int, error)
+	listFilesInBuild(job string, buildNumber int, prefix string) ([]string, error)
+	getFileFromBucket(job string, buildNumber int, path string) ([]byte, error)
+}
+
+// GCSBucketUtil that creates a Google Cloud Storage client to fetch data
+type GCSBucketUtil struct {
```

Comment:
One handy trick when you want a compile-time check that a type implements an interface is to declare an empty variable of $INTERFACE where the value is an empty instance of the type. Just drop this above the implementing type

```suggestion
var _ BucketUtil = &GCSBucketUtil{}

// GCSBucketUtil that creates a Google Cloud Storage client to fetch data
type GCSBucketUtil struct {
```